SocialCareCaseViewerApi.Tests/V1/Gateways/DatabaseGateway.Tests.cs

```diff
@@ -277,13 +277,40 @@ public void SelectAllocationsByWorkerEmail()
 
             _classUnderTest.CreateAllocation(createAllocationRequest);
 
-            var allocations = _classUnderTest.SelectAllocations(0, 0, worker.Email);
+            var allocations = _classUnderTest.SelectAllocations(0, 0, worker.Email, 0);
 
             allocations.Count.Should().Be(1);
             allocations.Single().AllocatedWorkerTeam.Should().Be(workerTeam.Team.Name);
             allocations.Single().AllocatedWorker.Should().Be($"{worker.FirstName} {worker.LastName}");
         }
 
+        [Test]
+        public void SelectAllocationsByTeamId()
+        {
+            // Create worker and teams
+            var worker = TestHelpers.CreateWorker(hasAllocations: false, hasWorkerTeams: false, id: 123);
+            var workerTeam = TestHelpers.CreateWorkerTeam(worker.Id);
+            worker.WorkerTeams = new List<WorkerTeam> { workerTeam };
+
+            DatabaseContext.Workers.Add(worker);
+            DatabaseContext.WorkerTeams.Add(workerTeam);
+            DatabaseContext.SaveChanges();
+
+            var (createAllocationRequest, _, allocator, resident, _) = TestHelpers.CreateAllocationRequest(workerId: worker.Id, teamId: workerTeam.TeamId);
+            DatabaseContext.Workers.Add(allocator);
+            DatabaseContext.Persons.Add(resident);
+            DatabaseContext.SaveChanges();
+
+            _classUnderTest.CreateAllocation(createAllocationRequest);
```

Comment:
Maybe worth adding couple of allocations for unrelated team IDs just to make sure it's not returning all of them.

Reply:
Done in [083ff24](https://github.com/LBHackney-IT/social-care-case-viewer-api/pull/610/commits/083ff24539601d24cbd2e28d4d957c53b695b337)